plugins/discovery-ec2/src/test/java/org/elasticsearch/cloud/aws/AwsEc2ServiceImplTests.java

```diff
@@ -0,0 +1,182 @@
+import com.amazonaws.auth.EnvironmentVariableCredentialsProvider;
+import com.amazonaws.auth.InstanceProfileCredentialsProvider;
+import com.amazonaws.auth.SystemPropertiesCredentialsProvider;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+
+public class AwsEc2ServiceImplTests extends ESTestCase {
+
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/19556")
+    public void testAWSCredentialsWithSystemProviders() {
+        AWSCredentialsProvider credentialsProvider = AwsEc2ServiceImpl.buildCredentials(logger, Settings.EMPTY);
+
+        AWSCredentials credentials = credentialsProvider.getCredentials();
+        assertThat(credentials.getAWSAccessKeyId(), is("DUMMY_ACCESS_KEY"));
+        assertThat(credentials.getAWSSecretKey(), is("DUMMY_SECRET_KEY"));
+    }
+
+    public void testAWSCredentialsWithElasticsearchAwsSettings() {
+        Settings settings = Settings.builder()
+            .put(AwsEc2Service.KEY_SETTING.getKey(), "aws_key")
+            .put(AwsEc2Service.SECRET_SETTING.getKey(), "aws_secret")
+            .build();
+        launchAWSCredentialsWithElasticsearchSettingsTest(settings, "aws_key", "aws_secret");
+    }
+
+    public void testAWSCredentialsWithElasticsearchEc2Settings() {
+        Settings settings = Settings.builder()
+            .put(AwsEc2Service.CLOUD_EC2.KEY_SETTING.getKey(), "ec2_key")
+            .put(AwsEc2Service.CLOUD_EC2.SECRET_SETTING.getKey(), "ec2_secret")
+            .build();
+        launchAWSCredentialsWithElasticsearchSettingsTest(settings, "ec2_key", "ec2_secret");
+    }
+
+    public void testAWSCredentialsWithElasticsearchAwsAndEc2Settings() {
+        Settings settings = Settings.builder()
+            .put(AwsEc2Service.KEY_SETTING.getKey(), "aws_key")
+            .put(AwsEc2Service.SECRET_SETTING.getKey(), "aws_secret")
+            .put(AwsEc2Service.CLOUD_EC2.KEY_SETTING.getKey(), "ec2_key")
+            .put(AwsEc2Service.CLOUD_EC2.SECRET_SETTING.getKey(), "ec2_secret")
+            .build();
+        launchAWSCredentialsWithElasticsearchSettingsTest(settings, "ec2_key", "ec2_secret");
+    }
+
+    protected void launchAWSCredentialsWithElasticsearchSettingsTest(Settings settings, String expectedKey, String expectedSecret) {
+        AWSCredentialsProvider credentialsProvider = AwsEc2ServiceImpl.buildCredentials(logger, settings);
+
+        try {
+            AWSCredentials credentials = credentialsProvider.getCredentials();
+            assertThat(credentials.getAWSAccessKeyId(), is(expectedKey));
+            assertThat(credentials.getAWSSecretKey(), is(expectedSecret));
+        } catch (AmazonClientException e) {
+            fail("We should not have raised an exception as we have credentials");
```

Comment:
You are throwing away the stack trace here. Just have this method throw Exception, and the tests that call it as well.